### ISSUE TITLE
p-overlayPanel: aria-expanded and aria-controls not added to trigger element

### ISSUE THREAD

**Issue by MaSch0212** [primefaces/primeng#14621]
### Describe the bug

When opening a p-overlayPanel element using a button, the `aria-expanded` and `aria-controls` attributes are not set to the button. If I understand the documentation correctly, this should happen.

> OverlayPanel adds aria-expanded state attribute and aria-controls to the trigger so that the relation between the trigger and the popup is defined.

I can reproduce the same issue with the `p-menu` in popup mode. I guess it might be the same issue.

### Environment

I can reproduce this on [primeng.org](https://primeng.org/overlaypanel).

### Reproducer

https://primeng.org/overlaypanel

### Angular version

17.0.2

### PrimeNG version

17.3.3

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.3.0

### Browser(s)

Chrome 120

### Steps to reproduce the behavior

1. Go to https://primeng.org/overlaypanel
2. In the basic example click the `Show` button
3. Check in DevTools if the aria attributes are added to that button
   ![image](https://github.com/primefaces/primeng/assets/8530764/64859805-6faf-4fc7-a17a-2961f9cea075)

### Expected behavior

- When showing the overlay, either on the `p-button` or `button` element,...
   - the `aria-expanded` attribute should be set to `true`
   - the `aria-controls` attribute should be set to the id of the overlay
- After closing the overlay...
   - the `aria-expanded` attribute is either removed or set to `false`
   - the `aria-controls` attribute is removed

**Comment by natebundy** [comment]
Seems like they're planning on leaving implementing this up to the user and the documentation is currently incorrect.

See https://github.com/primefaces/primeng/issues/14493

**Comment by mehmetcetin01140** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
